### PR TITLE
feat(client): add Timeout config

### PR DESCRIPTION
**Pull request by jtheoof**
This was a recommendation sent from the Meshify team.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,6 +76,15 @@ func main() {
 	}
 	m.Compression = comp
 
+	// Update the default timeout.
+	// The default Go http client will hang until a request has been fulfilled
+	// (potentially leaving the client hanging if the server is having issues).
+	// This setting allows to set the timeout for the client.
+	// Be careful that, when using Search module, some queries can take longer to return.
+	// Updating the timeout to a longer duration is advised for some queries.
+	m.Timeout = time.Second * 5
```
Comment (blemoine) [inline README.md]: why is it assigned twice ?

Comment (jtheoof) [reply README.md]: it's just an example for queries to restitution. I'll remove that if it's confusing.

Comment (jtheoof) [reply README.md]: Removed

**Comment by daddykotex** [comment]
Can you have a per-request timeout? Because a client-wide request seems like an important constraint...

**Comment by jtheoof** [comment]
@daddykotex This is a per-request timeout. Each time we create a new Http request client, it will release if the http call does not return after this timeout, for each request.

**Comment by daddykotex** [comment]
I understood the code like this: you set a global timeout contraint. Each request will respect this timeout (independently of other requests).

My question was more: with one client, can you send a request to restitution and ask for a four minute timeout and send a request to ingestion and ask for a 15s timeout for example

**Comment by jtheoof** [comment]
As discussed, let's keep the current client configuration and see with usage if we improve that part later on.

**Pull request merged**